Mathlib/NumberTheory/LSeries/QuadraticNonvanishing.lean

```diff
@@ -0,0 +1,188 @@
+a nontrivial (`χ ≠ 1`) quadratic (`χ^2 = 1`) Dirichlet character, then the value of
+its L-function at `s = 1` is nonzero.
+
+This is an important step in the proof of
+*Dirichlet's Theorem on Primes in Arithmetic Progression*.
+-/
+
+namespace DirichletCharacter
+
+/-!
+### Convolution of a Dirichlet character with ζ
+
+We define `DirichletCharacter.zetaMul χ` to be the arithmetic function obtained by
+taking the product (as arithmetic functions = Dirichlet convolution) of the
+arithmetic function `ζ` with `χ`.
+
+We then show that for a quadratic character `χ`, this arithmetic function is multiplicative
+and takes nonnegative real values.
+-/
+
+open Complex ArithmeticFunction
+
+variable {N : ℕ}
+
+/-- The complex-valued arithmetic function that is the convolution of the constant
+function `1` with `χ`. -/
+def zetaMul (χ : DirichletCharacter ℂ N) : ArithmeticFunction ℂ :=
+  .zeta * toArithmeticFunction (χ ·)
+
+/-- The arithmetic function `zetaMul χ` is multiplicative. -/
+lemma isMultiplicative_zetaMul (χ : DirichletCharacter ℂ N) : χ.zetaMul.IsMultiplicative :=
+  isMultiplicative_zeta.natCast.mul <| isMultiplicative_toArithmeticFunction χ
+
+lemma LSeriesSummable_zetaMul (χ : DirichletCharacter ℂ N) {s : ℂ} (hs : 1 < s.re) :
+    LSeriesSummable χ.zetaMul s := by
+  refine ArithmeticFunction.LSeriesSummable_mul (LSeriesSummable_zeta_iff.mpr hs) <|
+    LSeriesSummable_of_bounded_of_one_lt_re (m := 1) (fun n hn ↦ ?_) hs
+  simpa only [toArithmeticFunction, coe_mk, hn, ↓reduceIte, ← Complex.norm_eq_abs]
+  using norm_le_one χ _
+
+-- We use the ordering on `ℂ` given by comparing real parts for fixed imaginary part
+open scoped ComplexOrder
+
+lemma zetaMul_prime_pow_nonneg {χ : DirichletCharacter ℂ N} (hχ : χ ^ 2 = 1) {p : ℕ}
+    (hp : p.Prime) (k : ℕ) :
+    0 ≤ zetaMul χ (p ^ k) := by
+  simp only [zetaMul, toArithmeticFunction, coe_zeta_mul_apply, coe_mk,
+    Nat.sum_divisors_prime_pow hp, pow_eq_zero_iff', hp.ne_zero, ne_eq, false_and, ↓reduceIte,
+    Nat.cast_pow, map_pow]
+  rcases MulChar.isQuadratic_iff_sq_eq_one.mpr hχ p with h | h | h
+  · refine Finset.sum_nonneg fun i _ ↦ ?_
+    simp only [h, le_refl, pow_nonneg]
+  · refine Finset.sum_nonneg fun i _ ↦ ?_
+    simp only [h, one_pow, zero_le_one]
+  · simp only [h, neg_one_geom_sum]
+    split_ifs
+    exacts [le_rfl, zero_le_one]
+
+/-- `zetaMul χ` takes nonnegative real values when `χ` is a quadratic character. -/
+lemma zetaMul_nonneg {χ : DirichletCharacter ℂ N} (hχ : χ ^ 2 = 1) (n : ℕ) :
+    0 ≤ zetaMul χ n := by
+  rcases eq_or_ne n 0 with rfl | hn
+  · simp only [ArithmeticFunction.map_zero, le_refl]
+  · simpa only [χ.isMultiplicative_zetaMul.multiplicative_factorization _ hn] using
+      Finset.prod_nonneg
+        fun p hp ↦ zetaMul_prime_pow_nonneg hχ (Nat.prime_of_mem_primeFactors hp) _
+
+
+/-!
+### "Bad" Dirichlet characters
+
+Our goal is to show that `L χ 1 ≠ 0` when `χ` is a (nontrivial) quadratic Dirichlet character.
```

Comment:
```suggestion
Our goal is to show that `L(χ, 1) ≠ 0` when `χ` is a (nontrivial) quadratic Dirichlet character.
```
I think maybe here it's better to write `L(χ, 1)` since in docstring at L174 you already write `L(χ, 1)`.